dartagnan/src/test/java/com/dat3m/dartagnan/compilation/AbstractCompilationTest.java

```diff
@@ -82,47 +75,33 @@ protected Provider<Integer> getTimeoutProvider() {
     protected final Provider<Arch> sourceProvider = getSourceProvider();
     protected final Provider<Arch> targetProvider = getTargetProvider();
     protected final Provider<String> filePathProvider = () -> path;
-    protected final Provider<Integer> timeoutProvider = getTimeoutProvider();
     protected final Provider<Program> program1Provider = Providers.createProgramFromPath(filePathProvider);
     protected final Provider<Program> program2Provider = Providers.createProgramFromPath(filePathProvider);
     protected final Provider<Wmm> wmm1Provider = Providers.createWmmFromArch(getSourceProvider());
     protected final Provider<Wmm> wmm2Provider = Providers.createWmmFromArch(getTargetProvider());
     protected final Provider<EnumSet<Property>> propertyProvider = Provider.fromSupplier(() -> EnumSet.of(Property.getDefault()));
-    protected final Provider<VerificationTask> task1Provider = Provider.fromSupplier(() ->
-        VerificationTask.builder()
-        .withConfig(Configuration.builder()
-            .setOption(INITIALIZE_REGISTERS,String.valueOf(DO_INITIALIZE_REGISTERS))
-            .build())
-        .withTarget(sourceProvider.get())
-        .withSolverTimeout(timeoutProvider.get())
-        .build(program1Provider.get(), wmm1Provider.get(), propertyProvider.get()));
-    protected final Provider<VerificationTask> task2Provider = Provider.fromSupplier(() ->
-    	VerificationTask.builder()
-    	.withConfig(Configuration.builder()
-    		.setOption(INITIALIZE_REGISTERS,String.valueOf(DO_INITIALIZE_REGISTERS))
-    		.build())
-    	.withTarget(targetProvider.get())
-    	.withSolverTimeout(timeoutProvider.get())
-    	.build(program2Provider.get(), wmm2Provider.get(), propertyProvider.get()));
+    protected final Provider<VerificationTask> task1Provider = createTask(program1Provider, wmm1Provider, propertyProvider, sourceProvider, () -> 1);    
+    protected final Provider<VerificationTask> task2Provider = createTask(program2Provider, wmm2Provider, propertyProvider, targetProvider, () -> 1); 
```

Comment:
Don't you need to set the option `INITIALIZE_REGISTERS` anymore? What was the purpose of this before?
If you keep it like this, I would use the qualified name `Providers.createTask` instead of using a static import.